openedx/features/enterprise_support/enrollments/utils.py

```diff
@@ -38,32 +95,73 @@ def lms_enroll_user_in_course(
     Arguments:
      - username (str): User name
      - course_id (obj) : Course key obtained using CourseKey.from_string(course_id_input)
-     - mode (CourseMode): course mode
-     - enterprise_uuid (str): id to identify the enterprise to enroll under
-     - is_active (bool): Optional. A Boolean value that indicates whether the
+     - desired_mode (CourseMode): desired course mode
+     - is_active (bool): A Boolean value that indicates whether the
         enrollment is to be set to inactive (if False). Usually we want a True if enrolling anew.
+     - enterprise_uuid (str): Optional. id to identify the enterprise to enroll under
 
     Returns: A serializable dictionary of the new course enrollment. If it hits
      `CourseEnrollmentExistsError` then it logs the error and returns None.
     """
     user = _validate_enrollment_inputs(username, course_id)
+    current_enrollment = enrollment_api.get_enrollment(username, str(course_id))
 
     with transaction.atomic():
         try:
-            response = enrollment_api.add_enrollment(
+            if current_enrollment:
+                if (
+                    current_enrollment['mode'] != desired_mode
+                    or current_enrollment['is_active'] != is_active
+                ):
```

Comment:
nit: you can invert this logic and do an early-return to save a level of indentation, i.e.
```python
if not (blah blah blah):
    return None
response = update_enrollment(...)
```